src/cronjobs/productsCron.ts

```diff
@@ -0,0 +1,151 @@
+const cron = require('node-cron');
+const nodemailer = require('nodemailer');
+let transporter = nodemailer.createTransport({
+  host: 'smtp.gmail.com',
+  port: 587,
+  secure: false,
+  auth: {
+    user: process.env.EMAIL,
+    pass: process.env.PASSWORD,
+  },
+});
+
+const dateToMillseconds = (dateString: String) => {
+  const inputDate = new Date(`${dateString}T00:00:00`); // Convert to date object at midnight
+  const milliseconds = inputDate.getTime();
+  return milliseconds;
+};
+
+const millisecondsToDate = (milliseconds: any) => {
+  const date = new Date(milliseconds);
+  const year = date.getFullYear();
+  const month = date.getMonth() + 1;
+  const day = date.getDate();
+  const formattedDate = `${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`;
+  return formattedDate;
+};
+
+const products = [
+  {
+    name: 'mango',
+    owner: 'christian',
+    price: 10,
+    expirationDate: dateToMillseconds('2024-05-04'),
+    description: 'This is my products',
+    expired: false,
+  },
+  {
+    name: 'avocado',
+    owner: 'christian',
+    price: 15,
+    expirationDate: dateToMillseconds('2024-05-06'),
+    description: 'This is my products',
+    expired: false,
+  },
+];
+const names = [
+  {
+    name: 'christian',
+    age: 21,
+    email: 'christianinja3@gmail.com',
+  },
+  { name: 'kesh', age: 23, email: 'iradukundafabrice@gmail.com' },
+];
+
+const productExpireTask = cron.schedule('*/2 * * * * *', () => {
+  products.forEach(async (product) => {
+    // @ts-ignore
+    const expireDate = product.expirationDate;
+    const currentDate = Date.now();
+    // @ts-ignore
+    if (expireDate < currentDate) {
+      if (!product.expired) {
+        const ownerName = names.find((user) => user.name == product.owner);
+        product.expired = true;
+        sendExpiredEmail(ownerName, product);
+      }
+    }
+  });
+});
+
+const sendExpiredEmail = (receiver: any, product: any) => {
+  let date = millisecondsToDate(product.expirationDate);
+  let mailOptions = {
+    from: '"Sender Name" <christianinja3@gmail.com>', // sender address
+    to: `${receiver.email}`, // list of receivers
+    subject: `Product Expire`, // Subject line
+    text: 'Hello world?', // plain text body
+    html: `
```

Comment:
Can you move this html template in src/helps/MailTemplate.ts as it is on develop
<img width="1106" alt="image" src="https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn/assets/97307891/d86be9eb-1133-4fdf-adce-7b99cccc8531">